MueLu: compile error using OneAPI

## Bug Report
In nightly builds on Aurora, Trilinos builds are failing with the error 

```
unscoped enum 'lumpingType' requires fixed underlying type
```

This started on September 4, could be related to the merge of #14351.


### Description

<details>

<summary>error</summary>

```
/opt/aurora/24.180.3/updates/oneapi/compiler/eng-20240629/bin/compiler/../../include/sycl/handler.hpp:1725:59: error: unscoped enum 'lumpingType' requires fixed underlying type
 1725 |       h->kernel_parallel_for<TypesToForward..., Props...>(Args...);
      |                                                           ^
/opt/aurora/24.180.3/updates/oneapi/compiler/eng-20240629/bin/compiler/../../include/sycl/handler.hpp:1811:29: note: in instantiation of function template specialization 'sycl::handler::KernelPropertiesUnpackerImpl<>::kernel_parallel_for_unpack<Kokkos::Impl::FunctorWrapperRangePolicyPara
llelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, sycl::nd_item<>, Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>>' requested here
 1811 |           Unpacker.template kernel_parallel_for_unpack<KernelName, ElementType,
      |                             ^
/opt/aurora/24.180.3/updates/oneapi/compiler/eng-20240629/bin/compiler/../../include/sycl/handler.hpp:1807:5: note: in instantiation of function template specialization 'sycl::handler::unpack<Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, sycl::ext::oneapi::experimental::properties<std::tuple<>>, false, (lambda at /opt/aurora/24.180.3/updates/oneapi/compiler/eng-20240629/bin/compiler/../../include/sycl/handler.hpp:1810:21)>' requested here
 1807 |     unpack<KernelName, KernelType, PropertiesT,
      |     ^
/opt/aurora/24.180.3/updates/oneapi/compiler/eng-20240629/bin/compiler/../../include/sycl/handler.hpp:1480:5: note: in instantiation of function template specialization 'sycl::handler::kernel_parallel_for_wrapper<Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, sycl::nd_item<>, Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, sycl::ext::oneapi::experimental::properties<std::tuple<>>>' requested here
 1480 |     kernel_parallel_for_wrapper<NameT, TransformedArgType, KernelType,
      |     ^
/opt/aurora/24.180.3/updates/oneapi/compiler/eng-20240629/bin/compiler/../../include/sycl/handler.hpp:2506:5: note: in instantiation of function template specialization 'sycl::handler::parallel_for_impl<Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, 1, sycl::ext::oneapi::experimental::properties<std::tuple<>>>' requested here
 2506 |     parallel_for_impl<KernelName>(Range, Properties, std::move(KernelFunc));
      |     ^
/opt/aurora/24.180.3/updates/oneapi/compiler/eng-20240629/bin/compiler/../../include/sycl/handler.hpp:2591:5: note: in instantiation of function template specialization 'sycl::handler::parallel_for<Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, sycl::ext::oneapi::experimental::properties<std::tuple<>>, 1>' requested here
 2591 |     parallel_for<KernelName>(Range,
      |     ^
/lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/kokkos/core/src/SYCL/Kokkos_SYCL_ParallelFor_Range.hpp:113:13: note: in instantiation of function template specialization 'sycl::handler::parallel_for<Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>>, 1, Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>, Kokkos::RangePolicy<int, Kokkos::SYCL>> &>' requested here
  113 |         cgh.parallel_for<
      |             ^
/lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/kokkos/core/src/SYCL/Kokkos_SYCL_ParallelFor_Range.hpp:170:25: note: in instantiation of function template specialization 'Kokkos::Impl::ParallelFor<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::RangePolicy<int, Kokkos::SYCL>>::sycl_direct_launch<Kokkos::Impl::SYCLFunctionWrapper<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::Impl::SYCLInternal::USMObjectMem<sycl::usm::alloc::host>>>' requested here
  170 |     sycl::event event = sycl_direct_launch(m_policy, functor_wrapper,
      |                         ^
/lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/kokkos/core/src/Kokkos_Parallel.hpp:146:11: note: in instantiation of member function 'Kokkos::Impl::ParallelFor<MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, Kokkos::RangePolicy<int, Kokkos::SYCL>>::execute' requested here
  146 |   closure.execute();
      |           ^
/lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_kokkos_def.hpp:510:19: note: in instantiation of function template specialization 'Kokkos::parallel_for<Kokkos::RangePolicy<int, Kokkos::SYCL>, MueLu::MatrixConstruction::PointwiseFillNoReuseFunctor<KokkosSparse::CrsMatrix<double, int, Kokkos::Device<Kokkos::SYCL, Kokkos::SYCLDeviceUSMSpace>>, MueLu::MatrixConstruction::diag_lumping>, void>' requested here
  510 |           Kokkos::parallel_for("MueLu::CoalesceDrop::Fill_lumped_noreuse", range, fillFunctor);
```

</details>

--- a/packages/muelu/src/Graph/MatrixTransformation/MueLu_MatrixConstruction.hpp
+++ b/packages/muelu/src/Graph/MatrixTransformation/MueLu_MatrixConstruction.hpp
@@ -311,9 +311,9 @@ class PointwiseFillReuseFunctor {
     }
   }
 };
-enum lumpingType { no_lumping,
-                   diag_lumping,
-                   distributed_lumping };
+enum lumpingType : int { no_lumping,
+                         diag_lumping,
+                         distributed_lumping };
 
 /*!
   @class PointwiseFillNoReuseFunctor

